scan.go

```diff
@@ -167,6 +173,13 @@ func newScanCommand() *cobra.Command {
 	cmd.Flags().DurationVarP(&malwareAnalysisTimeout, "malware-analysis-timeout", "", 5*time.Minute,
 		"Timeout for malicious package analysis")
 
+	// mark code-db flag required only when dependencyUsageEvidence is enabled
+	cmd.PreRun = func(cmd *cobra.Command, args []string) {
+		if dependencyUsageEvidence {
```

Comment:
If we are making `--code` as the arg to enable code analysis during `vet scan` then I don't think we need to validate this. Too many args validation makes user experience complicated. I think we should just keep options to enable / disable larger modules (e.g. code analysis) and provide opinionated defaults.